Enable touchpad inertial scrolling(fix touchpad scrolls too fast)

seflerZ opened the pull request:
Close #150 @metalefty 
The xrdpMouse module doesn't handle touchpad inertial scrolling properly. At first I was intented to write a touchpad driver, now I found it uncessary. Because the RDP protocol doesn't have a touchpad channel, it should be good to add some touchpad function in the xrdpMouse driver.

Now the scrolling works great both on vertical and horizontal scrollings. Inertial scrolling also supported.
![touchpad-scroll_Trim](https://user-images.githubusercontent.com/2093588/189598314-a9e4f21f-5572-436b-b47c-f2f84d5c41f4.gif)

_Note: Some applications such as Firefox which uses WebKitGTK, have their own inertial handling process, so it won't work on every applications._

seflerZ committed:
toucpad scrolling works now

seflerZ committed:
refactored

seflerZ committed:
fix incorrect horizontal scrolling direction

metalefty commented:
Thanks, CI is failing but that's fine. This PR depends on neutrinolabs/xrdp#2364.

seflerZ committed:
Revert "mouse: workaround for too fast vertical scroll"

This reverts commit 6e136ac8ffb4fdeba5e610bd2475d1db59175611.

seflerZ commented:
@metalefty I've rebased the codes. Since the driver functions have been added. I deleted those hack codes at the same time.

BTW: Is there any tool or coding style guidance to pass the CI code style check?

matt335672 commented:
Hi @seflerZ

coding standard is [on the wiki](/neutrinolabs/xrdp/wiki/Coding-Style).

Alternatively, there's a script [run_astyle.sh](/neutrinolabs/xrdp/blob/devel/scripts/run_astyle.sh) in the [xrdp/scripts](/neutrinolabs/xrdp/tree/devel/scripts) directory that formats all your code for you using the `astyle` utility.

I can see that this PR won't compile until neutrinolabs/xrdp#2364 is merged.

Hope that's useful.

seflerZ commented:
@matt335672 Got it. I've fixed the code style issues in https://github.com/neutrinolabs/xrdp/pull/2364.

seflerZ committed:
fix code style issues

matt335672 commented:
@metalefty - I'm unable to re-run the CI checks on this PR for some reason. Can you?

metalefty commented:
Re-ran CI jobs.

matt335672 commented:
LGTM. @metalefty?

metalefty commented:
LGTM, too.

seflerZ commented:
Thanks for code review. @metalefty & @matt335672

furrnace commented:
starting from which release version is this fix included in xrdp?
